UniWatch/Services/EmailService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Configuration;
+using Microsoft.AspNet.Identity;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+using UniWatch.Enumerations;
+
+namespace UniWatch.Services
+{
+    /// <summary>
+    /// Represents the email client for SendGrid
+    /// </summary>
+    public class EmailService : IIdentityMessageService
+    {
+        private readonly SendGridAPIClient _service;
+
+        /// <summary>
+        /// The constructor for the email service
+        /// </summary>
+        /// <param name="apiKey">
+        /// The API key for the SendGrid account
+        /// </param>
+        public EmailService(string apiKey)
+        {
+            _service = new SendGridAPIClient(apiKey);
+        }
+
+        /// <summary>
+        /// The default constructor for the email service.
+        /// Uses the API key in the web config for the SendGrid API
+        /// </summary>
+        public EmailService() : this(WebConfigurationManager.AppSettings["SendGridApiKey"])
+        {       
+        }
+
+        /// <summary>
+        /// Sends an email
+        /// </summary>
+        /// <param name="from">
+        /// The email address that the recipient will respond to
+        /// </param>
+        /// <param name="to">
+        /// The email address of the recipient
+        /// </param>
+        /// <param name="subject">
+        /// The subject line of the email
+        /// </param>
+        /// <param name="body">
+        /// The contents of the email
+        /// </param>
+        /// <param name="type">
+        /// The type of the contents (plain, html, etc.)
+        /// </param>
+        /// <returns>
+        /// The result of the task
+        /// </returns>
+        public async Task<dynamic> SendEmail(string from, string to, string subject, string body, EmailContentType type=null)
```

Comment:
The parameter would need to be a compile time constant.

Have any ideas on how to get around this?